jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -516,7 +516,7 @@ public void doStart() throws Exception
 
 
         //Create the scheduler and start it
-        _scheduler = new ScheduledExecutorScheduler("Scanner-" + SCANNER_IDS.getAndIncrement(), true, 1);
+        _scheduler = new TimerScheduler("Scanner-" + SCANNER_IDS.getAndIncrement(), true);
```

Comment:
We want to move away from `Timer` as it's not configurable, has problems, etc. so this needs to go back to `ScheduledExecutorScheduler`.

Test case?

Reply:
The problem is that the ScheduledExecutorScheduler doesn't work: it interrupts its threads and throws the error you originally reported. The TimerScheduler doesn't manage its threads in that fashion and works just fine.

Reply:
This is not fixed yet.

Reply:
Changed to ScheduledExecutorScheduler - IMHO it's a bit ugly, but ....

I also think if you don't want TimerScheduler used, the class needs to be deprecated.